src/OpenLoco/src/World/CompanyAi.cpp

```diff
@@ -74,15 +81,168 @@ namespace OpenLoco
         company.var_85F0 = 0;
     }
 
+    enum class PurchaseVehicleResult
+    {
+        success = 0, // But maybe not all required purchases
+        allVehiclesPurchased = 1,
+        failure = 2
+    };
+
     // 0x00486ECF
-    static uint8_t sub_486ECF(Company& company, AiThought& thought)
+    static PurchaseVehicleResult purchaseVehicle(Company& company, AiThought& thought)
     {
-        // some sort of purchase vehicle
-        registers regs;
-        regs.esi = X86Pointer(&company);
-        regs.edi = X86Pointer(&thought);
-        call(0x00486ECF, regs);
-        return regs.al;
+        if ((company.challengeFlags & CompanyFlags::bankrupt) != CompanyFlags::none)
+        {
+            return PurchaseVehicleResult::failure;
+        }
+
+        if (thought.numVehicles >= thought.var_43)
+        {
+            return PurchaseVehicleResult::allVehiclesPurchased;
+        }
+
+        EntityId trainHeadId = EntityId::null;
+        for (auto i = 0; i < thought.var_45; ++i)
```

Comment:
can `var_45` be named?

Reply:
I like to leave naming of these until i'm really sure there aren't any other uses. I'm not 100% sure yet.